source/Cosmos.IL2CPU/AppAssembler.cs

```diff
@@ -870,6 +912,57 @@ public unsafe void GenerateVMTCode(HashSet<Type> aTypesSet, HashSet<MethodBase>
                 XS.Push(xDataName);
                 XS.Push(0);
 
+
+
+                // Enum entries
+                if (xType.IsEnum) { // If the type is an enum, insert actual useful information
+                    if(enumValues.Length != enumNames.Length) { // In **theory**, this is not possible
+                        throw new Exception("Type has unbalanced enum values/names data");
+                    }
+
+                    // This is kinda a compromise. We could surely change that in the future, somehow.
+                    // If this isn't fine and I should think of something else, tell me so when reviewing this pr :)
+
+                    // We only allow 32 byte long names so we can safely assume the size of the string
+                    foreach(var name in enumNames) {
+                        if (name.Length > 64) throw new Exception($"Enum names may not exceed 64 characters in length due to a technical limitation. Violating name: '{name}' in type {xType.FullName}");
+                    }
+
+                    xData = AllocateEmptyArray(enumValues.Length*2, sizeof(uint), xArrayTypeID);
+                    // Enum entries count
+                    XS.Push((uint)enumValues.Length);
+                    // Enum value entries
+                    xDataName = $"____SYSTEM____TYPE___{xTypeName}__EnumValuesArray";
+                    XSharp.Assembler.Assembler.CurrentInstance.DataMembers.Add(new DataMember(xDataName, xData));
+                    XS.Push(xDataName);
+                    XS.Push(0);
+
+                    xData = AllocateEmptyArray(enumValues.Length*16, sizeof(uint), xArrayTypeID);
+                    // Enum value entries
+                    xDataName = $"____SYSTEM____TYPE___{xTypeName}__EnumValueNamesArray";
+                    XSharp.Assembler.Assembler.CurrentInstance.DataMembers.Add(new DataMember(xDataName, xData));
+                    XS.Push(xDataName);
+                    XS.Push(0);
+
+                    // Enum underlying type id
+                    XS.Push(aGetTypeID(xType.GetEnumUnderlyingType()));
+                } else { // Otherwise, fill the data with pure emptiness. Just like my soul.
+                    xData = AllocateEmptyArray(0, sizeof(uint), xArrayTypeID);
```

Comment:
We should be able to include the data straight in this array here and save us the calls later.